### PR TITLE
Add a test for external links

### PR DESCRIPTION
**Pull request by Jeffxz**
First time writing some test. Checked with epubcheck and see no errors.

Kind of feel that open in a new browser instance might not be accurate if that strictly means open a new web browser. It can just open in a new tab (window) if it's web based reading system. Maybe we would like to add some note in the rs spec.

### CODE REVIEW

File: tests/pub-external-links/EPUB/package.opf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <dc:description>
+        Reading Systems SHOULD open links that resolve outside the EPUB Publication in a new browser instance
+        to ensure that the security and privacy controls of the browser are available to users.
+    </dc:description>
+    <dc:identifier id="pub-id">pub-external-links</dc:identifier>
+    <dc:language>en</dc:language>
+    <dc:title>External Links should be opened a new browser instance</dc:title>
+    <link rel="dcterms:rights" href="https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document"/>
+    <link rel="dcterms:rightsHolder" href="https://www.w3.org"/>
+    <meta property="dcterms:isReferencedBy">https://www.w3.org/TR/epub-rs-33/#sec-epub-rs-external-links</meta>
```
Comment (iherman) [inline tests/pub-external-links/EPUB/package.opf:16]: Per [the spec](https://www.w3.org/TR/epub-rs-33/#sec-epub-rs-external-links) this feature is a SHOULD and not a MUST, ie, the relevant meta element must be added.

```suggestion
    <meta property="belongs-to-collection">should</meta>
    <meta property="dcterms:isReferencedBy">https://www.w3.org/TR/epub-rs-33/#sec-epub-rs-external-links</meta>
```

File: tests/pub-external-links/EPUB/content_001.xhtml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops" xml:lang="en">
+  <head>
+    <title>TODO: Concise test title</title>
```
Comment (iherman) [inline tests/pub-external-links/EPUB/content_001.xhtml:3]:
```suggestion
    <title>External Links should be opened a new browser instance</title>
```

File: tests/pub-external-links/EPUB/nav.xhtml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops" xml:lang="en">
+  <head>
+    <title>TODO: Concise test title</title>
```
Comment (iherman) [inline tests/pub-external-links/EPUB/nav.xhtml:3]:
```suggestion
    <title>External Links should be opened a new browser instance</title>
```

**Review by iherman — CHANGES_REQUESTED** [review]

File: tests/pub-external-links/EPUB/package.opf (hunk 1 above)
Comment (Jeffxz) [reply tests/pub-external-links/EPUB/package.opf:16]: thanks for pointing out. will fix.

**Comment by Jeffxz** [comment]
thanks for review. updated.

**Review by iherman — APPROVED** [review]

File: tests/pub-external-links/EPUB/content_001.xhtml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops" xml:lang="en">
+  <head>
+    <title>External Links should be opened a new browser instance</title>
```
Comment (dlazin) [inline tests/pub-external-links/EPUB/content_001.xhtml:3]:
```suggestion
    <title>External links open a new browser instance</title>
```

File: tests/pub-external-links/EPUB/content_001.xhtml [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+  </head>
+  <body>
+    <p>Test passes if following link is opened in a new browser instance.</p>
```
Comment (dlazin) [inline tests/pub-external-links/EPUB/content_001.xhtml:6]:
```suggestion
    <p>Test passes if the following link opens in a new browser instance.</p>
```

Comment (dlazin) [inline tests/pub-external-links/EPUB/content_001.xhtml:6]: How does a tester *tell* that the external link opens in a new browser instances? Like, what are the characteristics?

File: tests/pub-external-links/EPUB/nav.xhtml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops" xml:lang="en">
+  <head>
+    <title>External Links should be opened a new browser instance</title>
```
Comment (dlazin) [inline tests/pub-external-links/EPUB/nav.xhtml:3]:
```suggestion
    <title>External links open a new browser instance</title>
```

File: tests/pub-external-links/EPUB/package.opf [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    <dc:coverage>Publication Resources</dc:coverage>
+    <dc:creator>Zheng Xu</dc:creator>
+    <dc:date>2022-03-06</dc:date>
+    <dc:description>
+        Reading Systems SHOULD open links that resolve outside the EPUB Publication in a new browser instance
+        to ensure that the security and privacy controls of the browser are available to users.
+    </dc:description>
+    <dc:identifier id="pub-id">pub-external-links</dc:identifier>
+    <dc:language>en</dc:language>
+    <dc:title>External Links should be opened a new browser instance</dc:title>
```
Comment (dlazin) [inline tests/pub-external-links/EPUB/package.opf:13]:
```suggestion
    <dc:title>External links open a new browser instance</dc:title>
```

File: tests/pub-external-links/EPUB/package.opf [hunk 8]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<package xmlns="http://www.idpf.org/2007/opf" xmlns:epub="http://www.idpf.org/2007/ops" version="3.0" xml:lang="en" unique-identifier="pub-id">
+  <metadata xmlns:dc="http://purl.org/dc/elements/1.1/">
+    <dc:coverage>Publication Resources</dc:coverage>
+    <dc:creator>Zheng Xu</dc:creator>
+    <dc:date>2022-03-06</dc:date>
+    <dc:description>
+        Reading Systems SHOULD open links that resolve outside the EPUB Publication in a new browser instance
+        to ensure that the security and privacy controls of the browser are available to users.
```
Comment (dlazin) [inline tests/pub-external-links/EPUB/package.opf:9]: To match the style of other examples at https://w3c.github.io/epub-tests/#sec-publication-resources-data, I'd say something like this:

The XHTML content document contains a link to an external website. The reading system should open it in a new browser instance.

**Review by dlazin — APPROVED** [review]
Sorry for the slow review, but this looks great! Please remember to make the corresponding change to the spec, pointing back to this test (the data-tests attribute).

File: tests/pub-external-links/EPUB/content_001.xhtml (hunk 5 above)
Comment (Jeffxz) [reply tests/pub-external-links/EPUB/content_001.xhtml:6]: yeah, good question. I would image either a new browser tab or a new browser window.

File: tests/pub-external-links/EPUB/package.opf (hunk 8 above)
Comment (Jeffxz) [reply tests/pub-external-links/EPUB/package.opf:9]: thanks. Let me make the change.

**Comment by Jeffxz** [comment]
Thanks @iherman. I made change of test files and regenerated epub. Will send a PR to epub33 later today

File: tests/pub-external-links/EPUB/content_001.xhtml (hunk 4 above)
Comment (Jeffxz) [reply tests/pub-external-links/EPUB/content_001.xhtml]: thnx, fixed

File: tests/pub-external-links/EPUB/content_001.xhtml (hunk 5 above)
Comment (Jeffxz) [reply tests/pub-external-links/EPUB/content_001.xhtml]: updated

File: tests/pub-external-links/EPUB/nav.xhtml (hunk 6 above)
Comment (Jeffxz) [reply tests/pub-external-links/EPUB/nav.xhtml]: updated

File: tests/pub-external-links/EPUB/package.opf (hunk 7 above)
Comment (Jeffxz) [reply tests/pub-external-links/EPUB/package.opf]: thnx, fixed

**Comment by Jeffxz** [comment]
> Sorry for the slow review, but this looks great! Please remember to make the corresponding change to the spec, pointing back to this test (the data-tests attribute).

thanks. I will send PR to spec later today. Something like this commit, right? https://github.com/w3c/epub-specs/commit/ff3a57ec33eac009afa397faab6e2b59d6a1dee3

**Comment by dlazin** [comment]
> > Sorry for the slow review, but this looks great! Please remember to make the corresponding change to the spec, pointing back to this test (the data-tests attribute).
> 
> thanks. I will send PR to spec later today. Something like this commit, right? [w3c/epub-specs@ff3a57e](https://github.com/w3c/epub-specs/commit/ff3a57ec33eac009afa397faab6e2b59d6a1dee3)

Yes, just like that!

**Comment by Jeffxz** [comment]
ok, thanks @dlazin here is the PR https://github.com/w3c/epub-specs/pull/2088 to spec

**Pull request merged**